src/userscores/view/exportscoremodel.cpp

```diff
@@ -0,0 +1,363 @@
+{
+    if (!index.isValid()) {
+        return QVariant();
+    }
+
+    INotationPtr notation = m_notations[index.row()];
+
+    switch (role) {
+    case RoleTitle:
+        return notation->metaInfo().title;
+    case RoleIsSelected:
+        return m_selectionModel->isSelected(index);
+    case RoleIsMain:
+        return isMainNotation(notation);
+    }
+
+    return QVariant();
+}
+
+int ExportScoreModel::rowCount(const QModelIndex&) const
+{
+    return m_notations.size();
+}
+
+QHash<int, QByteArray> ExportScoreModel::roleNames() const
+{
+    static const QHash<int, QByteArray> roles {
+        { RoleTitle, "title" },
+        { RoleIsSelected, "isSelected" },
+        { RoleIsMain, "isMain" }
+    };
+
+    return roles;
+}
+
+void ExportScoreModel::toggleSelection(int scoreIndex)
+{
+    if (!isNotationIndexValid(scoreIndex)) {
+        return;
+    }
+
+    QModelIndex modelIndex = index(scoreIndex);
+    m_selectionModel->QItemSelectionModel::select(modelIndex, QItemSelectionModel::Toggle);
+
+    emit dataChanged(modelIndex, modelIndex);
+}
+
+void ExportScoreModel::toggleAllSelections(bool select)
+{
+    QModelIndexList previousSelectedIndexes = m_selectionModel->selectedIndexes();
+    m_selectionModel->QItemSelectionModel::select(QItemSelection(index(0), index(
+                                                                     m_notations.size() - 1)),
+                                                  select ? QItemSelectionModel::Select : QItemSelectionModel::Deselect);
+    QModelIndexList newSelectedIndexes = m_selectionModel->selectedIndexes();
+
+    QSet<QModelIndex> indexesToUpdate(previousSelectedIndexes.begin(), previousSelectedIndexes.end());
+    indexesToUpdate = indexesToUpdate.unite(QSet<QModelIndex>(newSelectedIndexes.begin(), newSelectedIndexes.end()));
+
+    for (const QModelIndex& indexToUpdate : indexesToUpdate) {
+        emit dataChanged(indexToUpdate, indexToUpdate);
+    }
+}
+
+IMasterNotationPtr ExportScoreModel::masterNotation() const
+{
+    return context()->currentMasterNotation();
+}
+
+bool ExportScoreModel::isMainNotation(INotationPtr notation) const
+{
+    return notation == masterNotation()->notation();
+}
+
+bool ExportScoreModel::isNotationIndexValid(int index) const
+{
+    return index >= 0 && index < m_notations.size();
+}
+
+QList<int> ExportScoreModel::selectedRows() const
+{
+    QList<int> result;
+
+    for (const QModelIndex& index: m_selectionModel->selectedIndexes()) {
+        result << index.row();
+    }
+
+    return result;
+}
+
+int ExportScoreModel::getSelectionLength() const
+{
+    return selectedRows().size();
+}
+
+QString ExportScoreModel::getExportPath() const
+{
+    return m_exportPath.toQString();
+}
+
+void ExportScoreModel::setExportPath(QString path)
+{
+    m_exportPath = io::path(path);
+}
+
+io::path ExportScoreModel::getDefaultExportPath() const
+{
+    return configuration()->scoresPath() + "/" + masterNotation()->metaInfo().title + ".pdf";
```

Comment:
1. Better to add a method to the configuration that will generate a required path from a given title:

IUserScoresConfiguration {
io::path defaultExportPath(title)
...
}

2. We don't use the "get" prefix in getters. It will just defaultExportPath